### PR TITLE
Ankit | Test | Prod - State dropdown is not getting open when GSTIN has been removed from address

### PR DESCRIPTION
**Pull request by ankitdubeyoffice**
https://app.clickup.com/t/86cw2y5dv

### CODE REVIEW

File: apps/web-giddh/src/assets/css/global/input-dropdown-field.scss [hunk 1]
```diff
@@ -66,6 +66,21 @@ report-filters,
             border-color: var(--color-fields-border) !important;
             border-left: none !important;
         }
+        
+        &:has(input:disabled, input:read-only) {
```
Comment (nisha-walkover) [inline apps/web-giddh/src/assets/css/global/input-dropdown-field.scss]: already have a class for disabled

**Review by nisha-walkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/shared/create-address/create-address.component.ts [hunk 2]
```diff
@@ -516,11 +516,10 @@ export class CreateAddressComponent implements OnInit, OnDestroy {
      * @memberof CreateAddressComponent
      */
     public isReadonly(): boolean {
-        const isEditAddress = this.addressConfiguration?.type === 'editAddress';
-        const hasTaxNumber = this.addressToUpdate?.taxNumber?.length > 0;
         const isGSTIN = this.addressConfiguration?.tax?.name === 'GSTIN';
         const stateLabelNotNull = this.addressForm?.get('stateLabel')?.value !== null;
-        const taxNumberNotNull = this.addressForm?.get('taxNumber')?.value !== null;
-        return (isEditAddress && hasTaxNumber && isGSTIN) || (stateLabelNotNull && taxNumberNotNull);
+        const taxNumberNotEmpty = this.addressForm?.get('taxNumber')?.value !== "";
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/shared/create-address/create-address.component.ts]:
```suggestion
        const taxNumberNotEmpty = this.addressForm?.get('taxNumber')?.value !== "" && this.addressForm?.get('taxNumber')?.value !== null;
```

File: apps/web-giddh/src/app/shared/create-address/create-address.component.ts [hunk 3]
```diff
@@ -516,11 +516,10 @@ export class CreateAddressComponent implements OnInit, OnDestroy {
...
+        const taxNumberNotNull = this.addressForm?.get('taxNumber')?.value !== null;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/shared/create-address/create-address.component.ts:522]:
```suggestion
   
```

File: apps/web-giddh/src/app/shared/create-address/create-address.component.ts [hunk 4]
```diff
@@ -516,11 +516,10 @@ export class CreateAddressComponent implements OnInit, OnDestroy {
...
     public isReadonly(): boolean {
-        const isEditAddress = this.addressConfiguration?.type === 'editAddress';
-        const hasTaxNumber = this.addressToUpdate?.taxNumber?.length > 0;
         const isGSTIN = this.addressConfiguration?.tax?.name === 'GSTIN';
         const stateLabelNotNull = this.addressForm?.get('stateLabel')?.value !== null;
-        const taxNumberNotNull = this.addressForm?.get('taxNumber')?.value !== null;
-        return (isEditAddress && hasTaxNumber && isGSTIN) || (stateLabelNotNull && taxNumberNotNull);
+        const taxNumberNotEmpty = this.addressForm?.get('taxNumber')?.value !== "";
+        const taxNumberNotNull = this.addressForm?.get('taxNumber')?.value !== null;      
+        return  (isGSTIN && stateLabelNotNull && taxNumberNotNull && taxNumberNotEmpty);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/shared/create-address/create-address.component.ts]:
```suggestion
        return  isGSTIN && stateLabelNotNull && taxNumberNotEmpty;
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/shared/create-address/create-address.component.ts (hunk 2 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/shared/create-address/create-address.component.ts]: done

File: apps/web-giddh/src/app/shared/create-address/create-address.component.ts (hunk 4 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/shared/create-address/create-address.component.ts]: done

**Review by dvCodeWorld — APPROVED** [review]

**Review by dvCodeWorld — APPROVED** [review]

**Review by nisha-walkover — APPROVED** [review]

**Pull request merged**